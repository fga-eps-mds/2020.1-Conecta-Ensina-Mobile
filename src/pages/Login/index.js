import React, {useState, useEffect, useContext} from 'react';
import Theme, {theme} from '../../../Theme';
import Background2 from '../../components/Background2';
import CustomText from '../../components/CustomText';
<<<<<<< HEAD
import {AuthContext} from '../../contexts/auth';
=======
import RegFieldBig from '../../components/RegFieldBig';
import {AuthContext} from '../../contexts/auth'
>>>>>>> 2e5b79381e511676b1eacc73f6c6f9b5bf94b2d7
import {
  LoginContainer,
  ButtonAluno,
  ButtonProfessor,
  Container,
  ButtonEntrar,
  Link,
  LinkTexto,
  Logo,
  UserContatiner,
  Icon,
  ButtonAdm,
} from './styles';

export default function Login({navigation}) {
<<<<<<< HEAD
  const [selectedButton, setSelectedButton] = useState(null);
  const [background1, setBackground1] = useState(null);
  const [background2, setBackground2] = useState(null);
  const [background3, setBackground3] = useState(null);
  const {userSelected} = useContext(AuthContext);

  useEffect(() => {
    if (selectedButton === 'Aluno') {
      setBackground1(theme.colors.cinzaClaro);
      setBackground2(theme.colors.fundoAzul);
      setBackground3(theme.colors.fundoAzul);
    } else if (selectedButton === 'Professor') {
      setBackground1(theme.colors.fundoAzul);
      setBackground2(theme.colors.cinzaClaro);
      setBackground3(theme.colors.fundoAzul);
    } else if (selectedButton === 'Adm') {
      setBackground1(theme.colors.fundoAzul);
      setBackground2(theme.colors.fundoAzul);
      setBackground3(theme.colors.cinzaClaro);
    } else if (selectedButton === null) {
      setBackground1(theme.colors.fundoAzul);
      setBackground2(theme.colors.fundoAzul);
      setBackground3(theme.colors.fundoAzul);
    }
  }, [selectedButton]);

=======
  const [email, setEmail] = useState('')
  const [password, setPassword] = useState('')
  const {signIn} = useContext(AuthContext);

  function handleLogin(){
    signIn(email, password)
  }
  
>>>>>>> 2e5b79381e511676b1eacc73f6c6f9b5bf94b2d7
  return (
    <Theme>
      <Background2
        blue={<Logo source={require('../../assets/logo.png')} />}
        white={
          <Container>
            <LoginContainer>
              <UserContatiner>
                <Icon source={require('../../assets/user_white.png')} />
              </UserContatiner>
              <RegFieldBig
                    placeholder="Email"
                    autoCapitalize="none"
                    value={email}
                    onChangeText={ (text) => setEmail(text) }
              />
              <RegFieldBig
                    placeholder="Senha"
                    autoCapitalize="none"
                    value={password}
                    onChangeText={ (text) => setPassword(text) }
                    secureTextEntry={true}
              />
              <ButtonEntrar onPress={handleLogin}>
                <CustomText white medium>
                  Entrar
                </CustomText>
              </ButtonEntrar>
              <Link>
                <LinkTexto onPress={() => navigation.navigate('RegistroAluno')}>
                  Registre-se
                </LinkTexto>
              </Link>
            </LoginContainer>
<<<<<<< HEAD
            <ButtonAluno
              onPress={() => {
                setSelectedButton('Aluno');
              }}
              style={{backgroundColor: background1}}>
              <CustomText white medium>
                Aluno
              </CustomText>
            </ButtonAluno>
            <ButtonProfessor
              onPress={() => {
                setSelectedButton('Professor');
              }}
              style={{backgroundColor: background2}}>
              <CustomText white medium>
                Professor
              </CustomText>
            </ButtonProfessor>
            <ButtonAdm
              onPress={() => {
                setSelectedButton('Adm');
              }}
              style={{backgroundColor: background3}}>
              <CustomText white medium>
                Adm
              </CustomText>
            </ButtonAdm>
=======
>>>>>>> 2e5b79381e511676b1eacc73f6c6f9b5bf94b2d7
          </Container>
        }
      />
    </Theme>
  );
}
