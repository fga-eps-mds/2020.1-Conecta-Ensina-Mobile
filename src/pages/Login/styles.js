import styled from 'styled-components';
import {ButtonGeral} from '../../components/ButtonGeral/styles';

export const LoginContainer = styled.View`
  background-color: ${(props) => props.theme.colors.cinzaClaro};
  height: 230px;
  width: 300px;
  border-radius: 15px;
  align-items: center;
  justify-content: center;
  margin-top: 55.5%;
`;

export const ButtonAluno = styled(ButtonGeral)`
  margin-top: 11.2%;
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const ButtonProfessor = styled(ButtonGeral)`
  align-self: center;
  margin-top: 11.2%
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const Container = styled.View`
  align-items: center;
  margin-bottom: 83.5%;
`;

export const ButtonEntrar = styled(ButtonGeral)`
  align-items: center;
`;

export const Link = styled.TouchableOpacity`
  margin-bottom: 10%;
`;

export const LinkTexto = styled.Text`
  font-family: ${(props) => props.theme.font};
  font-size: ${(props) => props.theme.fontSizes.smallMedium};
  color: ${(props) => props.theme.colors.branco};
  text-shadow: -1px 1px 8px ${(props) => props.theme.colors.fundoAzul};
`;

export const Logo = styled.Image`
  width: 50%;
  height: 76.5%;
  margin-bottom: 14%;
`;

export const UserContatiner = styled.View`
  width: 19.5%;
  height: 25.5%;
  border-radius: 100px;
  background-color: ${(props) => props.theme.colors.fundoAzul};
  margin-bottom: 10px;
  align-items: center;
  justify-content: center;
`;

export const Icon = styled.Image`
  width: 52%;
  height: 63%;
`;
