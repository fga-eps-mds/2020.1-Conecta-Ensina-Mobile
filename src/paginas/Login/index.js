import React from 'react';
import Theme from '../../../Theme';
import Background2 from '../../components/Background2';
import CustomText from '../../components/CustomText';
import {
  LoginContainer,
  ButtonAluno,
  Texto,
  ButtonProfessor,
  Container,
  ButtonEntrar,
  Link,
  LinkTexto,
  Logo,
  UserContatiner,
  Icon,
} from './styles';

export default function Login() {
  return (
    <Theme>
      <Background2
        blue={<Logo source={require('../../assets/logo.png')} />}
        white={
          <Container>
            <LoginContainer>
              <UserContatiner>
                <Icon source={require('../../assets/user_white.png')} />
              </UserContatiner>
              <Link>
                <LinkTexto>Registre-se</LinkTexto>
              </Link>
              <ButtonEntrar>
                <CustomText white medium>
                  Entrar
                </CustomText>
              </ButtonEntrar>
            </LoginContainer>
            <ButtonAluno>
              <CustomText white medium>
                Aluno
              </CustomText>
            </ButtonAluno>
            <ButtonProfessor>
              <CustomText white medium>
                Professor
              </CustomText>
            </ButtonProfessor>
          </Container>
        }
      />
    </Theme>
  );
}
