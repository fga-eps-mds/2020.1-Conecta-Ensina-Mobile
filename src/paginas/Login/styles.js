import styled from 'styled-components';
import {ButtonGeral} from '../../components/ButtonGeral/styles';

export const LoginContainer = styled.View`
  background-color: ${(props) => props.theme.colors.cinzaClaro};
  height: 230px;
  width: 300px;
  border-radius: 15px;
  align-items: center;
  justify-content: center;
`;

export const ButtonAluno = styled(ButtonGeral)`
  margin-top: 40px;
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const Texto = styled.Text`
  font-family: ${(props) => props.theme.font};
  font-size: ${(props) => props.theme.fontSizes.medium};
  color: ${(props) => props.theme.colors.branco};
`;

export const ButtonProfessor = styled(ButtonGeral)`
  align-self: center;
  margin-top: 40px;
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const Container = styled.View`
  align-items: center;
  margin-bottom: 300px;
`;

export const ButtonEntrar = styled(ButtonGeral)`
  align-items: center;
`;

export const Link = styled.TouchableOpacity`
  margin-bottom: 30px;
`;

export const LinkTexto = styled.Text`
  font-family: ${(props) => props.theme.font};
  font-size: 20px;
  color: ${(props) => props.theme.colors.branco};
  text-shadow: -1.5px 1.5px 4px #184e79;
`;

export const Logo = styled.Image`
  width: 201px;
  height: 208px;
  margin-bottom: 10px;
`;

export const UserConatiner = styled.View`
  width: 58px;
  height: 58px;
  border-radius: 100px;
  background-color: ${(props) => props.theme.colors.fundoAzul};
  margin-bottom: 10px;
`;
