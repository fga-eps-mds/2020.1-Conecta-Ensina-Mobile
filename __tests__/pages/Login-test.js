import '@testing-library/jest-native/extend-expect';
import React, {useContext} from 'react';
import Routes from '../../src/routes/index';
import Login from  '../../src/pages/Login/index';
import {render, fireEvent} from '@testing-library/react-native';
import {act} from 'react-test-renderer';
import { NavigationContainer } from '@react-navigation/native';
import AuthProvider from '../../src/contexts/auth.js'

//Silence warning
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');


describe(('testing Login page'),()=>{

  test(('should renders background'), ()=>{
    const {getByTestId} = render(<Login/>);
    const background = getByTestId('background');

    expect(background).toBeDefined();
  });


  test(('should renders logo'), ()=>{
    const {getByTestId} = render(<Login/>);
    const logo = getByTestId('Logo');

    expect(logo.props.source).toEqual({ testUri: '../../../src/assets/logo.png' });
  });


  test(('should renders Icon'), ()=>{
    const {getByTestId} = render(<Login/>);
    const icon = getByTestId('Icon');

    expect(icon.props.source).toEqual({ testUri: '../../../src/assets/user_white.png' });
  });


  test(('button "Registrar-se" should be pressable'), async ()=>{
    const component = (
      <NavigationContainer>
        <AuthProvider>
          <Routes/>
        </AuthProvider>
      </NavigationContainer>  
    );
  
    const {getByText} = render(component);
    await act(async ()=>{
      const RegistrarButton = getByText('Registre-se');
      const EntrarButton = await getByText('Entrar');
  
      fireEvent.press(RegistrarButton);
      //Por enquanto o teste abaixo nÃ£o funciona
      fireEvent.press(EntrarButton);
    })

    
  });
  test(('should be match snapshot'),()=>{
    const page = render(<Login/>).toJSON();

    expect(page).toMatchSnapshot();
  })
}); 

