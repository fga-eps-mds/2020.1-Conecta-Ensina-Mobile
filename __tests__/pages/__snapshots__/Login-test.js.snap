// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Login page should be match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#184E79",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
  testID="background"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#184E79",
          "flexBasis": 0,
          "flexGrow": 0.7,
          "flexShrink": 1,
          "justifyContent": "center",
          "width": "100%",
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/logo.png",
        }
      }
      style={
        Array [
          Object {
            "height": "76.5%",
            "marginBottom": "14%",
            "width": "50%",
          },
        ]
      }
      testID="Logo"
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderTopLeftRadius": 20,
          "borderTopRightRadius": 20,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "marginBottom": "83.5%",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#D9D9D9",
              "borderRadius": 15,
              "height": 230,
              "justifyContent": "center",
              "marginTop": "55.5%",
              "width": 300,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#184E79",
                "borderRadius": 100,
                "height": "25.5%",
                "justifyContent": "center",
                "marginBottom": 10,
                "width": "19.5%",
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/user_white.png",
              }
            }
            style={
              Array [
                Object {
                  "height": "63%",
                  "width": "52%",
                },
              ]
            }
            testID="Icon"
          />
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "marginBottom": "10%",
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontFamily": "Roboto",
                  "fontSize": 20,
                  "textShadowColor": "#184E79",
                  "textShadowOffset": Object {
                    "height": 1,
                    "width": -1,
                  },
                  "textShadowRadius": 8,
                },
              ]
            }
          >
            Registre-se
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#184E79",
              "borderRadius": 10,
              "height": 50,
              "justifyContent": "center",
              "opacity": 1,
              "width": 200,
            }
          }
        >
          <Text
            medium={true}
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontFamily": "Roboto",
                  "fontSize": 25,
                },
              ]
            }
            white={true}
          >
            Entrar
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#184E79",
            "borderRadius": 10,
            "display": "flex",
            "height": 50,
            "justifyContent": "center",
            "marginTop": "11.2%",
            "opacity": 1,
            "width": 200,
          }
        }
      >
        <Text
          medium={true}
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Roboto",
                "fontSize": 25,
              },
            ]
          }
          white={true}
        >
          Aluno
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#184E79",
            "borderRadius": 10,
            "display": "flex",
            "height": 50,
            "justifyContent": "center",
            "marginTop": "11.2%",
            "opacity": 1,
            "width": 200,
          }
        }
      >
        <Text
          medium={true}
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Roboto",
                "fontSize": 25,
              },
            ]
          }
          white={true}
        >
          Professor
        </Text>
      </View>
    </View>
  </View>
</View>
`;
